src/d/actor/d_a_hmlif.cpp

```diff
@@ -38,110 +74,462 @@ static dCcD_SrcSph l_sph_src = {
 void daHmlif_c::set_next_pnt() {
-    /* Nonmatching */
+    s8 uVar1 = m448;
+    BOOL iVar5 = FALSE;
+    if (m445 != 0xff) {
+        if (chkSwitchPathType()) {
+            int roomNo = fopAcM_GetRoomNo(this);
+            if (((ppd == dPath_GetRoomPath(m444, roomNo)) && (uVar1 == ppd->mArg0)) || (ppd != dPath_GetRoomPath(m444, roomNo) && (uVar1 == 0))) {
+                iVar5 = setNextPath();
+            }
+        }
+
+        if (!iVar5) {
+            m448 += m446;
+            dPath* pdVar2 = ppd;
+            if (dPath_ChkClose(ppd)) {
+                if (m448 > (s8)pdVar2->m_num - 1) {
+                    m448 = 0;
+                } else if (m448 < 0) {
+                    m448 = pdVar2->m_num - 1;
+                }
+            } else if (m448 > pdVar2->m_num - 1) {
+                m446 = -1;
+                m448 = ppd->m_num - 2;
+            } else if (m448 < 0) {
+                m446 = 1;
+                m448 = 1;
+            }
+        }
+
+        m45C = m450;
+        dPnt* pnt = &ppd->m_points[m448];
+        m450 = pnt->m_position;
+    }
 }
 
 /* 00001410-00001520       .text setNextPath__9daHmlif_cFv */
-void daHmlif_c::setNextPath() {
-    /* Nonmatching */
+BOOL daHmlif_c::setNextPath() {
+    int roomNo = fopAcM_GetRoomNo(this);
+    BOOL uVar5 = FALSE;
+    dPath* pdVar4 = ppd;
+    dPath* pInPath = dPath_GetRoomPath(m444, roomNo);
+    dPath* pdVar1 = dPath_GetNextRoomPath(pInPath, roomNo);
+
+    if (fopAcM_isSwitch(this, m474)) {
+        if (pdVar4 == pInPath) {
+            m447 = m446;
+        }
+        ppd = pdVar1;
+        m448 = 1;
+        m446 = 1;
+        uVar5 = TRUE;
+    } else if (ppd != pInPath) {
+        m445 = m444;
+        ppd = pInPath;
+        if (ppd->mArg0 == ppd->m_num) {
+            m447 = -1;
+        } else if (ppd->mArg0 == 0) {
+            m447 = 1;
+        }
+        m446 = m447;
+        m448 = ppd->mArg0 + m446;
+        uVar5 = TRUE;
+    }
+    return uVar5;
 }
 
 /* 00001520-00001534       .text chkSwitchPathType__9daHmlif_cFv */
-void daHmlif_c::chkSwitchPathType() {
-    /* Nonmatching */
+bool daHmlif_c::chkSwitchPathType() {
+    return m474 != 0xFF;
 }
 
 /* 00001534-00001610       .text Draw__9daHmlif_cFv */
 BOOL daHmlif_c::Draw() {
-    /* Nonmatching */
+    g_env_light.settingTevStruct(TEV_TYPE_ACTOR, &current.pos, &tevStr);
+    g_env_light.setLightTevColorType(mpModel, &tevStr);
+    if (mpBtp != NULL) {
+        mpBtp->entry(mpModel->getModelData(), m48A);
+    }
+
+    if (mpBrk != NULL) {
+        mpBrk->entry(mpModel->getModelData());
+    }
+
+    dComIfGd_setListBG();
+    mDoExt_modelUpdateDL(mpModel);
+    dComIfGd_setList();
+    return TRUE;
 }
 
 /* 00001610-00001630       .text daHmlif_Create__FPv */
-static cPhs_State daHmlif_Create(void* i_this) {
-    return ((daHmlif_c*)i_this)->daHmlifCreate();
+static cPhs_State daHmlif_Create(void* v_this) {
+    return ((daHmlif_c*)v_this)->daHmlifCreate();
```

Comment:
Does v_this mean something? It doesn't appear in the assertion strings for either TWW or TP.